openadapt/db/db.py

```diff
@@ -76,6 +76,7 @@ def get_base(engine: sa.engine) -> sa.engine:
 engine = get_engine()
 Base = get_base(engine)
 Session = sessionmaker(bind=engine)
+ReadOnlySession = sessionmaker(bind=engine, autoflush=False, autocommit=False)
```

Comment:
This appears to not be read-only, since the user can still write if they manually call `commit()` on the session. What do you think about naming it `NonLockingSession`, and renaming the arguments to get_new_session to be `non_locking` and `standard` or similar?

Reply:
Good idea, let's do it!

Reply:
As discussed, please add a test 🙏